### ISSUE TITLE
Add cross platform path support to workerDirectory

### ISSUE THREAD

**Issue by Tornek** [kettanaito/msw#2218]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 18 or higher

### Node.js version

20.15.1

### Reproduction repository

-

### Reproduction steps

To reproduce my problem you will need both a Mac and a Windows computer.
You need to write path with slashes in msw.workerDirectory in package.json. For example: 
```
package.json
{
  ...,
  "msw": {
    "workerDirectory": "public/sw"
  },
  ...
}
```
And run your application with msw

### Current behavior

On windows we get this message

![image](https://github.com/user-attachments/assets/42ba3aea-83c6-4cdd-b8a9-4e7ed1998171)

On Mac we have `Service Worker successfully created!`

### Expected behavior

We have `Service Worker successfully created!` on each computers

**Comment by nabby27** [comment]
Great project!

I'm gonna create a small bounty for this issue if you don't mind @kettanaito since I'll like to see this feature implemented

**Comment by opiredev** [comment]
@nabby27 created a $50.00 reward using [Opire](https://opire.dev) 
 <details><summary>How to earn this reward?</summary>

Since this project does not have Opire bot installed yet 😞 you need to go to [Opire](https://app.opire.dev) and claim the rewards through your programmer's dashboard once you have your PR ready 💪</details>

**Comment by mobley-trent** [comment]
I'd like to give this issue a go @kettanaito @nabby27

**Comment by ivanfernandez2646** [comment]
@Tornek I think you use an old version of the library (for the message that you show). 
Despite of is a minor problem, I'll try to solve it :)))

Keep in contact!

**Comment by ivanfernandez2646** [comment]
@nabby27 The bot here is not working, is it installed in the project??. I'm trying to start working on it...

**Comment by nabby27** [comment]
@ivanfernandez2646 Exactly, the Opire bot is not installed in this project but you can solve it anyway and claim the reward in your dashboard with the PR

**Comment by ivanfernandez2646** [comment]
@Tornek as I said in the last version it works correctly.

I test on Windows and MacOs computers...

Also I create a PR prove of that with some tescases that cover this.

Thanks!!

@kettanaito @nabby27

**Comment by kettanaito** [comment]
## Released: v2.4.2 🎉

This has been released in v2.4.2!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.4.2)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.4.2)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
